internal/validators/status/validator.go

```diff
@@ -177,7 +177,7 @@ func (sv *statusValidator) listCheckRunsForRef(ctx context.Context) ([]*github.C
 			return nil, err
 		}
 		runResults = append(runResults, cr.CheckRuns...)
-		if cr.GetTotal() < maxCheckRunsPerPage {
+		if cr.GetTotal() <= len(runResults) {
```

Comment:
Thanks for raising this! 🥰
In case of `getCombinedStatus`, the actual returned data of "Total count" seems to be the number of jobs in that particular page you are querying for, rather than the list of all the jobs associated with that particular git ref.

It is possible that we may still have some glitches in the current implementation, but let's get this fix in first, and work out other parts as a separate ticket if necessary 👍